### PR TITLE
Improve wording around Kata containers and OCI runtime spec

### LINKED ISSUE #40114: https://k8s.io/docs/concepts/security/multi-tenancy incorrectly lists Kata Containers as OCI compliant
From the docs..

> Kata Containers is an OCI compliant runtime that allows you to run containers in a VM. The hardware virtualization available in Kata offers an added layer of security for containers running untrusted code

As of version 2.0, Kata Containers no longer provides an OCI compliant runtime.

### PR DESCRIPTION
**Pull request by adityasamant25**
Fixes #40114 

From the Kubernetes documentation standpoint, a minor change in the wording should help clear the ambiguity specified in issue #40114. Instead of stating that Kata Containers is an OCI _"compliant"_ runtime, I'm proposing a change to state that Kata containers _"support"_ the OCI Runtime Specification (from v3.0.2 and above).

### CODE REVIEW

**Comment by sftim** [comment]
PR titles become commit messages

/retitle Improve wording around Kata containers and OCI runtime spec

File: content/en/docs/concepts/security/multi-tenancy.md [hunk 1]
```diff
@@ -291,9 +291,9 @@ sandboxing implementations are available:
 
 * [gVisor](https://gvisor.dev/) intercepts syscalls from containers and runs them through a
   userspace kernel, written in Go, with limited access to the underlying host.
-* [Kata Containers](https://katacontainers.io/) is an OCI compliant runtime that allows you to run
+* [Kata Containers](https://katacontainers.io/) provide a secure container runtime that allows you to run
   containers in a VM. The hardware virtualization available in Kata offers an added layer of
-  security for containers running untrusted code.
+  security for containers running untrusted code. [v3.0.2](https://github.com/kata-containers/kata-containers/releases/tag/3.0.2) and above support the OCI Runtime Specification.
```
Comment (dipesh-rawat) [inline content/en/docs/concepts/security/multi-tenancy.md:296]: This statement may require some evidence as I'm unsure about its validity.

The older release (https://github.com/kata-containers/kata-containers/releases/tag/2.5.2) mentions

> Kata Containers 2.5.2 support the OCI Runtime Specification [v1.0.0-rc5](https://github.com/opencontainers/runtime-spec/releases/tag/v1.0.0-rc5)


💭 Could we potentially omit this level of detail from here? People interested in the OCI compliance status of Kata Containers could easily check the Kata Containers website (https://katacontainers.io/), conveniently linked earlier in the text. What do you think?

Comment (adityasamant25) [reply content/en/docs/concepts/security/multi-tenancy.md]: I agree. This is addressed in the latest commit. Thanks @dipesh-rawat.

**Review by dipesh-rawat — COMMENTED** [review]
I feel this appears sufficient to eliminate ambiguity regarding OCI compliance as raised in the issue https://github.com/kubernetes/website/issues/40114. 

@kubernetes/sig-node-pr-reviews, please review the current proposal and provide any improvements.

/approve

**Comment by adityasamant25** [comment]
@dipesh-rawat Although approved, absence of LGTM label is blocking the merge of the PR. Can you add the LGTM as well?

**Pull request merged**